packages/core/lib/process-request.ts

```diff
@@ -210,23 +232,35 @@ export const processRequest = async <TContext = {}, TRootValue = {}>(
           // execution result.
           if (isAsyncIterable(result)) {
             const asyncIterator = result[Symbol.asyncIterator]();
+            const transformedIterator = {
+              ...asyncIterator,
+              next: async () => {
+                const iteratorResult = await asyncIterator.next();
+                if (iteratorResult.done) {
+                  return iteratorResult;
+                }
+                return formatPayload({
+                  payload: iteratorResult.value,
+                  context,
+                  rootValue,
+                  document,
+                  operation,
+                })
+              },
+            };
             return {
```

Comment:
shouldn't this object also have all other properties from AsyncGenerator (next, return, throw)?

Reply:
```ts
interface AsyncGenerator<T = unknown, TReturn = any, TNext = unknown> extends AsyncIterator<T, TReturn, TNext> {
    // NOTE: 'next' is defined using a tuple to ensure we report the correct assignability errors in all places.
    next(...args: [] | [TNext]): Promise<IteratorResult<T, TReturn>>;
    return(value: TReturn | PromiseLike<TReturn>): Promise<IteratorResult<T, TReturn>>;
    throw(e: any): Promise<IteratorResult<T, TReturn>>;
    [Symbol.asyncIterator](): AsyncGenerator<T, TReturn, TNext>;
}
```

Reply:
It is just AsyncIterable which is enough for "for await"